app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt

```diff
@@ -16,19 +16,20 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
 import com.google.accompanist.placeholder.placeholder
 import dev.johnoreilly.common.domain.entities.GameFixture
-import dev.johnoreilly.fantasypremierleague.presentation.FantasyPremierLeagueViewModel
 import dev.johnoreilly.fantasypremierleague.presentation.global.lowfidelitygray
 import dev.johnoreilly.fantasypremierleague.presentation.global.maroon200
 
 @Composable
 fun FixturesListView(
-    fantasyPremierLeagueViewModel: FantasyPremierLeagueViewModel,
     onFixtureSelected: (fixtureId: Int) -> Unit,
 ) {
-    val fixturesState = fantasyPremierLeagueViewModel.gameweekToFixtures.collectAsState()
-    val currentGameweek: State<Int> = fantasyPremierLeagueViewModel.currentGameweek.collectAsState()
+    val fixturesViewModel: FixturesViewModel = viewModel()
```

Comment:
A pattern I'm using in other projects (and should have done so here as well) is pass view model in as parameter (with default value of `getViewMode()` as is done in following for example https://github.com/joreilly/PeopleInSpace/blob/main/app/src/main/java/com/surrus/peopleinspace/personlist/PersonListScreen.kt#L60 ).

Reply:
This seems to be approach generally used in various compose samples as it makes it easier to test the composable.

Reply:
Yeah, I think that's preferable all right.  Often maybe be a top level screen composable might know about the viewmodel (or at least it's in constructor like that) and then pass only data down to lower level composables.  I like approach in PeopleInSpace (stolen from NowInAndroid!) where "route" composable does this.

Reply:
Yes that is exactly how it is done in Joe Birch's course too.

Reply:
Why doesn't that PeopleInSpace example throw an exception for not having a zero-argument-constructor for the ViewModel ?

Reply:
Which code is this?

Reply:
>  pass view model in as parameter (with default value of getViewMode() as is done in following for example https://github.com/joreilly/PeopleInSpace/blob/main/app/src/main/java/com/surrus/peopleinspace/personlist/PersonListScreen.kt#L60 )

[That VM takes in a repo](https://github.com/joreilly/PeopleInSpace/blob/8463a3c84e81c1b0988f2ca9d3363f92b4634119/app/src/main/java/com/surrus/peopleinspace/personlist/PersonListViewModel.kt#L20)

If I copy that though, I get an error for my VM having no zero-arg constructor. I might pull down PIS and compare

Reply:
that's injected by Koin (based on what's in `AppModule`